src/main/java/frc/robot/systems/EveryBotIntakeFSM.java

```diff
@@ -0,0 +1,340 @@
+						return EveryBotIntakeFSMState.IDLE_STOP;
+					}
+				}
+				return EveryBotIntakeFSMState.INTAKING;
+			case IDLE_STOP:
+				if (input.isOuttakeButtonPressed()
+					&& flipMotor.getEncoder().getPosition() > FLIP_THRESHOLD) {
+					return EveryBotIntakeFSMState.OUTTAKING;
+				} else if (input.isIntakeButtonPressed()
+					&& !(flipMotor.getEncoder().getPosition() > FLIP_THRESHOLD)) {
+					return EveryBotIntakeFSMState.INTAKING;
+				} else if (input.isFlipButtonPressed()) {
+					//&& arm.getEncoderCount() > BASE_THRESHOLD) {
+					if (flipMotor.getEncoder().getPosition() > FLIP_THRESHOLD) {
+						return EveryBotIntakeFSMState.IDLE_FLIPCOUNTERCLOCKWISE;
+					} else {
+						return EveryBotIntakeFSMState.IDLE_FLIPCLOCKWISE;
+					}
+				} else {
+					return EveryBotIntakeFSMState.IDLE_STOP;
+				}
+			case IDLE_FLIPCLOCKWISE:
+				if (flipMotor.getEncoder().getPosition() > FLIP_THRESHOLD
+					|| input.isFlipAbortButtonPressed()) {
+					return EveryBotIntakeFSMState.IDLE_STOP;
+				} else if (input.isFlipButtonPressed()) {
+					return EveryBotIntakeFSMState.IDLE_FLIPCOUNTERCLOCKWISE;
+				} else if (flipMotor.getEncoder().getPosition() < FLIP_THRESHOLD) {
+					return EveryBotIntakeFSMState.IDLE_FLIPCLOCKWISE;
+				}
+			case IDLE_FLIPCOUNTERCLOCKWISE:
+				if (flipMotor.getEncoder().getPosition() < 0
+					|| input.isFlipAbortButtonPressed()) {
+					return EveryBotIntakeFSMState.IDLE_STOP;
+				} else if (input.isFlipButtonPressed()) {
+					//&& arm.getEncoderCount() > BASE_THRESHOLD) {
+					return EveryBotIntakeFSMState.IDLE_FLIPCLOCKWISE;
+				} else {
+					return EveryBotIntakeFSMState.IDLE_FLIPCOUNTERCLOCKWISE;
+				}
+			case OUTTAKING:
+				if (input.isOuttakeButtonPressed()) {
+					return EveryBotIntakeFSMState.OUTTAKING;
+				} else {
+					return EveryBotIntakeFSMState.IDLE_STOP;
+				}
+			default:
+				throw new IllegalStateException("Invalid state: " + currentState.toString());
+		}
+	}
+
+	/* ------------------------ FSM state handlers ------------------------ */
+	/**
+	 * Handle behavior in states.
+	 */
+	private void handleIntakingState() {
+		spinnerMotor.set(INTAKE_SPEED);
+		flipMotor.set(0);
+	}
+	private void handleIdleStopState() {
+		spinnerMotor.set(KEEP_SPEED);
+		flipMotor.set(0);
+	}
+	private void handleFlipClockWiseState() {
+		flipMotor.set(-TURN_SPEED);
+		if (flipMotor.getEncoder().getPosition() >= FLIP_THRESHOLD) {
+			flipMotor.set(0);
+		}
+		spinnerMotor.set(0);
+	}
+	private void handleFlipCounterClockWiseState() {
+		flipMotor.set(TURN_SPEED);
+		if (flipMotor.getEncoder().getPosition() <= 0) {
+			flipMotor.set(0);
+		}
+		spinnerMotor.set(0);
+	}
+	private void handleOuttakingState(TeleopInput input) {
+		if (input == null) {
+			if (!hasTimerStarted) {
+				timer.reset();
+				timer.start();
+				hasTimerStarted = true;
+			}
+		}
+		for (int i = 0; i < AVERAGE_SIZE; i++) {
+			currLogs[i] = 0;
+		}
+		if (itemType == ItemType.CUBE) {
+			spinnerMotor.set(RELEASE_SPEED);
```

Comment:
Should we have different release speeds for cone vs cube?

Reply:
No